components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/ext/SessionManager.kt

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.tab.collections.ext
+
+import android.content.Context
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.tab.collections.Tab
+import mozilla.components.feature.tab.collections.TabCollection
+
+/**
+ * Restores the given [Tab] from a [TabCollection]. Will invoke [onTabRestored] on successful restore
+ * and [onFailure] otherwise.
+ */
+fun SessionManager.restore(
```

Comment:
These methods are not used yet, right? Should we add tests for it? We can also file a follow-up to cover them.

Reply:
They are getting used in this PR: https://github.com/mozilla-mobile/fenix/pull/9399

We only have on-device tests for this component (because of Room) and I noticed that they are failing and we didn't notice since they do not run in automation. I filed an issue to fix them. As part of that we could add tests for those methods too:
https://github.com/mozilla-mobile/android-components/issues/6380